zebra/zebra_rnh.c

```diff
@@ -1268,73 +1270,181 @@ int zebra_send_rnh_update(struct rnh *rnh, struct zserv *client,
 	return -1;
 }
 
-static void print_nh(struct nexthop *nexthop, struct vty *vty)
+static void print_nh(struct nexthop *nexthop, struct vty *vty,
+		     json_object *json)
 {
 	char buf[BUFSIZ];
 	struct zebra_ns *zns = zebra_ns_lookup(nexthop->vrf_id);
 
 	switch (nexthop->type) {
 	case NEXTHOP_TYPE_IPV4:
 	case NEXTHOP_TYPE_IPV4_IFINDEX:
-		vty_out(vty, " via %pI4", &nexthop->gate.ipv4);
-		if (nexthop->ifindex)
-			vty_out(vty, ", %s",
-				ifindex2ifname_per_ns(zns, nexthop->ifindex));
+		if (json) {
+			json_object_string_addf(json, "ip", "%pI4",
+						inet_ntoa(nexthop->gate.ipv4));
```

Comment:
oh, please use inet_ntop here.

Reply:
Just use the `json_object_string_addf` formatting? `pI4` is fine, no need for `inet_ntoa`.

Reply:
yes, better